frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/contextmenu/ShowSelectedBranchInGitLogAction.java

```diff
@@ -0,0 +1,107 @@
+package com.virtuslab.gitmachete.frontend.actions.contextmenu;
+
+import static com.virtuslab.gitmachete.frontend.resourcebundles.GitMacheteBundle.format;
+import static com.virtuslab.gitmachete.frontend.resourcebundles.GitMacheteBundle.getString;
+
+import java.util.Objects;
+import java.util.concurrent.CancellationException;
+import java.util.concurrent.ExecutionException;
+
+import com.intellij.openapi.actionSystem.AnActionEvent;
+import com.intellij.openapi.progress.ProgressIndicator;
+import com.intellij.openapi.progress.Task;
+import com.intellij.openapi.project.Project;
+import com.intellij.vcs.log.Hash;
+import com.intellij.vcs.log.impl.VcsLogContentUtil;
+import com.intellij.vcs.log.impl.VcsProjectLog;
+import git4idea.branch.GitBranchesCollection;
+import lombok.CustomLog;
+import lombok.val;
+import org.checkerframework.checker.guieffect.qual.UIEffect;
+
+import com.virtuslab.gitmachete.frontend.actions.base.BaseGitMacheteRepositoryReadyAction;
+import com.virtuslab.gitmachete.frontend.actions.expectedkeys.IExpectsKeySelectedBranchName;
+import com.virtuslab.logger.IEnhancedLambdaLogger;
+
+@CustomLog
+public class ShowSelectedBranchInGitLogAction extends BaseGitMacheteRepositoryReadyAction
+    implements
+      IExpectsKeySelectedBranchName {
+
+  @Override
+  public IEnhancedLambdaLogger log() {
+    return LOG;
+  }
+
+  @Override
+  @UIEffect
+  protected void onUpdate(AnActionEvent anActionEvent) {
+    super.onUpdate(anActionEvent);
+
+    val presentation = anActionEvent.getPresentation();
+    if (!presentation.isEnabledAndVisible()) {
+      return;
+    }
+
+    val selectedBranchName = getSelectedBranchName(anActionEvent);
+    // It's very unlikely that selectedBranchName is empty at this point since it's assigned directly before invoking this
+    // action in GitMacheteGraphTable.GitMacheteGraphTableMouseAdapter.mouseClicked; still, it's better to be safe.
+    if (selectedBranchName.isEmpty()) {
+      presentation.setEnabled(false);
+      presentation.setDescription(getString("action.GitMachete.ShowSelectedBranchInGitLogAction.undefined.branch-name"));
+      return;
+    }
+
+    presentation.setDescription(
+        format(getString("action.GitMachete.ShowSelectedBranchInGitLogAction.description.precise"), selectedBranchName.get()));
+  }
+
+  @Override
+  @UIEffect
+  public void actionPerformed(AnActionEvent anActionEvent) {
```

Comment:
AFAICS bincompat checks fails on `EXPERIMENTAL_API_USAGES`... TBH I wouldn't consider it a blocker, esp. if there isn't an easy (hack-free) way to circumvent (as long as that's the only class of bincompat-related problems). Consider modifying settings of `runPluginVerifier` in build.gradle if needed

Reply:
![image](https://user-images.githubusercontent.com/19799111/165268732-14dead1c-5222-4b59-b562-8e708cd17bc9.png)

Here is tthe other approach but it uses an experimental interface (VcsLogUiEx)

Reply:
Hmmm wait... but doesn't the **current approach** use some experimental API as well (`MainVcsLogUi`)? 🤔

Reply:
So either case (`MainVcsLogUi`/`VcsLogUiEx`) , we'd end up with `EXPERIMENTAL_API_USAGES` error anyway (unless suppressed) ?

Reply:
sadly :c

Reply:
Okay... and I suppose:
1. in case of `MainVcsLogUi`, we'll only have `EXPERIMENTAL_API_USAGES` on all supported versions but NOT any more serious errors like ClassNotFound,
2. in case of `VcsLogUiEx` we'll have `EXPERIMENTAL_API_USAGES` on some of latest supported version, and a showstopper bincompat error on earlier version

Amirite?

Reply:
yup, thats how I see it